### PR TITLE
Add blog post on Spyder's SciPy 2024 IDE UX BoF

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Adds a new blog post summarizing the discussion at and insights gained from Spyder's SciPy 2024 Birds of a Feather session, "What do you want to see from your scientific IDE", focusing on the UX that users are looking for from their interfaces for scientific computation.

### CODE REVIEW

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
summary: The Spyder team hosted a Birds of a Feather session at SciPy 2024, this time on the topic of users' experiences (good and bad) with the UI/UX of scientific interfaces and IDEs, and how their developers can better serve users. Here, we share what we learned from the session, as well as a link to the full detailed community notes.
```

I think `another` is meant to imply continuity with the BoF hosted in 2023, but for people that haven't read our previous blog posts it'll sound a bit weird. That's why I propose this change.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
The Spyder team hosted a Birds of a Feather (BoF) session, [What do you want to see from your scientific IDE?](https://cfp.scipy.org/2024/talk/VWMV7B/), at [SciPy 2024](https://www.scipy2024.scipy.org/).
```

Typo.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
In the session, we had on behalf of Spyder not only myself but also [Pierre Raybaut](https://pierreraybaut.github.io/), the original Spyder creator himself (who also gave a great [talk on Spyder's history and his current work](https://www.youtube.com/watch?v=MKPUuV1LEqk)), as well as another former Spyder developer.
```

- Avoid repeating `room` (used in the previous paragraph).
- Improve phrasing a bit to make the sentence a bit easier to grasp.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Participants empathized that scientific coding is different from other forms of programming, and that the "Copilots of the world" may not help with scientific aspects of coding.
```

Typo?

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Indeed, they would not necessarily be expected to assist properly designing valid digital experiments, correctly implementing appropriate scientific algorithms, ensuring the analysis techniques used are valid for the data, and choosing appropriate metrics and statistics to  interpret the results.
```

The sentence has more meaning for me like this.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Another participant stated that more so than citations for code, the most important thing AI would help with is tests that verify that the code does what it is supposed to do.
```

I haven't seen `averred` before.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
THey also expressed a desire for IDEs having native support for interactive map displays like in QGIS or ArcGIS rather than just static Matplolib images, which would make for more rapid prototyping.
```

This seems necessary.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md:59]:
```suggestion
Their solution supports any database type that Pandas does, with the simplest interface to it, so it feels like a CSV file, and would like IDEs to support that workflow.
```

Typo.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Finally, a PyCharm user shared how Xarray/Pandas data types, their extension system and data accessors were all opaque to its autocomplete, at least without further tagging or augmentation to the IDE / code.
```

Typo and `data types` seems missing (it took me several seconds to understand this sentence without that).

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Posing the question directly, we asked people to share their opinions on a web-based application vs. a desktop application, and on a "desktop" app using a webview as its UI.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
On a related note, an attendee shared that they support a small statistics group, and their biggest challenge is helping students learn scientific computing.
```

Typo?

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
This prompted others to relate how, where they work, platforms for accessing data require a lot of authentication, and entail using JupyterHub as their default IDE/platform, with very little flexibility in the matter.
```

Typo and the sentence reads much better to me like this.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
They would like a platform for other IDEs to be able to connect to a shared JupyterHub environment, which is exactly what we are working on for Spyder 6.1 and beyond.
```

- I don't think it's necessary to say that this confidential when we're making it public.
- Avoid repeating Spyder.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
One fairly specific topic that got a disproportionate amount of attention was incorporating better support for GPU computing (e.g. CUDA) within IDEs and interfaces, thanks to Nvidia representatives in the room posing the initial question.
```

`computing` sounds better to me but I don't have a strong preference about it.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
However, they said it would be nice to offer a workflow to shorten that development loop.
```

Typo.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: So this was a PI asking their own postdoc if they were getting the most perfomance out of their GPUs? If so, I think this sentence could be rephrased to make it clearer because it's a bit confusing as it is (I don't have a suggestion in mind).

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
The postdoc's reply, which the attendee found rather telling, was that "PyTorch [just] runs": it was painful enough for them to just get it running. so they were not particularly interested in spending even more time optimizing it further.
```

I think this makes the sentence flow better.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
They have students who don't know how to use the 4 CPUs in their laptop, much less the GPUs in a supercomputer center.
```

Since we're not talking about a specific center, I think using `a` is better.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
But, they said, it poses a real sociological challenge to get students used to using, much less optimizing, GPUs.
```

Seems missing.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
In reply, participants suggested better integration with NVSMI or an integrated dashboard of GPU usage, which would allow users to see that, for instance, a particular kernel crunching data is too big to use the full potential of the GPU.
```

Seems unnecessary (otherwise I don't understand the meaning of this sentence).

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
They mentioned that tooling in Python tends to be "pretty thin" as users often don't go looking for tools and use the ones that are there, whereas their profiler can be really intimidating and throws a lot of information at such users, potentially putting them off.
```

Remove blank.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
All told, there was a surprising amount of interest and discussion around this once-niche topic, signaling that GPU computing has become something much closer to a mainstream need in the scientific space.
```

Again, not strong preference about it but it sounds better to me.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
A particularly salient comment was one of the attendees' request that "We need RStudio for Python", with another replying that Spyder fills most of that gap.
```

Seems missing.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
By contrast, the most surprising was the extensive discussion about GPU computing and how IDEs and tools can help with that, which seems to indicate that IDE plugins for assisting in this regard might be valuable.
```

- `compute` -> `computing` to go with my previous suggestions.
- It's easier for me (as a non-native speaker) to grasp the meaning of this sentence with the second suggestion.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
While a subset of such could be built into an IDE (as we have with Spyder's Variable Explorer), given the diversity of scientific data types across fields this suggests the need for a more pluggable/extensible approach to variable exploration.
```

Sounds better to me.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
To that end, we're working to provide those facilities in Spyder 6.1, and could eventually be developed into such an common interface.
```

- In 6.0 we have our own custom `jupyter-server` to which users can connect only through ssh. 6.1 will have support to connect to any JupyterHub running instance.
- I'm still unsure about the `common interface` part, but we'll see how that goes when Hendrick adds that support in the coming months.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
Thanks to everyone for attending and sharing such great ideas and insights, and we look forward to continuing to learn from the community and hear more in the future.
```

To not imply that we're restricted to the attendees' feedback.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [inline src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: Perhaps we should add here the picture someone at NF took for the BoF.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @CAM-Gerlach for your work on this!

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: Yeah, where's that photo? I was so focused on leading the BoF that I (again) forgot to take photos, sorry 🤦

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]:
```suggestion
In addition to myself, attending the session we had [Pierre Raybaut](https://pierreraybaut.github.io/), the original creator of Spyder (who also gave a great [talk on Spyder's history and his current work](https://www.youtube.com/watch?v=MKPUuV1LEqk)), as well as another former Spyder developer.
```

Further phrasing fixes and improvements

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Just a informal shorthand, but should be expanded here, yeah—was a copy/paste mistake from the original notes. I also noticed an internal repetition/phrasing issue, so fixing both:

```suggestion
Participants empathized that scientific coding is different from other forms of programming, and that the "Copilots of the world" may not help with the former.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Yup, that's an editing error, although (ironically given the previous suggestion), "with" is missing:

```suggestion
Indeed, they would not necessarily be expected to assist with properly designing valid digital experiments, correctly implementing appropriate scientific algorithms, ensuring the analysis techniques used are valid for the data, and choosing appropriate metrics and statistics to  interpret the results.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: It's a fancy way to say "stated a point/counterpoint", which I used to avoid repeating "stated" in the previous sentence used in the same position. However, if its potentially confusing for less experienced or non-native speakers, we can replace it with something else:

```suggestion
Another participant added that more so than citations for code, the most important thing AI would help with is tests that verify that the code does what it is supposed to do.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:59]: The maintainers consider their name "pandas" to be canonically lowercase. See, e.g., the [main page of the pandas documentation](https://pandas.pydata.org/docs/index.html). Personally I think it a bit silly, but as a fellow PyData project I'm inclined to respect their wishes, at least in a context where a common noun would not otherwise be capitalized.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Was just transcribing the user's comment (as recorded in the notes), which didn't specify the data objects specifically and rather the packages themselves, though from my own experiance it must presumably be what they mean since autocomplete does generally work on module-level functions (unless something's broken with PyCharm). However, it does need some rewording, is Still missing a word and see the note on the casing change above. Also fixed some other issues with the sentance.

```suggestion
Finally, a PyCharm user shared how the xarray/pandas data types and their extension systems and data accessors were all opaque to that IDE's autocomplete, at least without further tagging or augmentation in the IDE or the code.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Thanks for the catch; also made some other improvements.

```suggestion
Posing the question directly, we asked people to share their opinions on a web-based vs. desktop applications, and on a true desktop programs versus those just with a webview as a UI.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Good improvements, thanks. Given the profusion of commas, it seemed much better to split into two sentences:

```suggestion
This prompted others to relate how, where they work, platforms for accessing data require a lot of authentication.
This entails using JupyterHub as their default IDE/platform, with very little flexibility in the matter.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Haha, "confidential" was a pure typo/editing mistake; have no idea how it got there—maybe I meant to type "coincidentally" and then autocorrected it to that?

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Yup and yup:

```suggestion
Another attendee supervises a postdoc using a bunch of A100 GPUs, and asked the postdoc how he knew he was getting the most performance out of it.
```

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md]: Yeah, though to use more appropriate syntax to avoid too many potentially confusing commas, and further clarify the sentance:

```suggestion
However, the attendee expressed concern that this poses a real sociological challenge to get students used to using—much less optimizing—GPUs.
```

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks Carlos for your great feedback! I've implemented all of the suggestions (save a couple minor ones) with some further fixes and improvements of my own.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: No worries. The photo is [here](https://x.com/Spyder_IDE/status/1813011498224263378) (I just took it from another NF post on X).

Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: Thanks; added

Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: Hmm, seems like its not displaying, even though AFAIK I've added the image in the same way as the other blog posts. Is there anything you can spot that I'm doing wrong?

**Comment by CAM-Gerlach** [comment]
Hey @conradolandia , I added an image to this blog post, but its inappropriately displaying the [alt text](https://spec.commonmark.org/0.31.2/#images) as a caption instead.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (CAM-Gerlach) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: Hmm, looks like its working now. But its displaying the alt text as a caption, which is not correct.

**Comment by CAM-Gerlach** [comment]
Besides the alt text issue, @ccordoba12 there is still [one comment](https://github.com/spyder-ide/spyder-website/pull/6#discussion_r1752293771) nominally awaiting your response or resolution (though as it should be resolved from my side and is very minor, I won't let it block merge).

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (ccordoba12) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:59]: Let's leave it as pandas then.

File: src/routes/blog/ide-ux-bof-scipy-2024/+page.md
Comment (conradolandia) [reply src/routes/blog/ide-ux-bof-scipy-2024/+page.md:9]: This has been solved by [PR 14](https://github.com/spyder-ide/spyder-website/pull/14). I will mark the conversation as resolved.

**Comment by CAM-Gerlach** [comment]
I updated it to the latest `main` once we merged #14 , and now the alt text is correctly treated as alt text, so this should be good to merge pending formal approval. The date is set for tomorrow (UTC), so should be good to merge anytime today or tomorrow (CDT/COT).

Also for some reason, GitHub glitched out and thinks @conradolandia reviewed this PR instead of @ccordoba12 ... when I selected re-review on @ccordoba12 's review it invited Carlos as a new reviewer and then showed the existing review as Andres'. Weird.

**Review by ccordoba12 — APPROVED** [review]
Great work @CAM-Gerlach!

**Pull request merged**
